### PR TITLE
bpo-29553: Fixed ArgumentParses format_usage for mutually exclusive groups

### PR DESCRIPTION
**Pull request by andrewnester**
Fix for https://bugs.python.org/issue29553

### CODE REVIEW

File: Lib/test/test_argparse.py [hunk 1]
```diff
@@ -2725,6 +2725,35 @@ def get_parser(self, required):
           -c          c help
         '''
 
+class TestMutuallyExclusiveNested(TestCase):
+    def test_format_usage(self):
+        parser = ErrorRaisingArgumentParser(prog='PROG')
```
Comment (berkerpeksag) [inline Lib/test/test_argparse.py]: Is there a reason you can't use the existing test infrastructure? I think it's possible to convert these tests to follow the style of `TestMutuallyExclusiveOptionalAndPositional`: https://github.com/python/cpython/blob/82e38bd80113ecfc7aff2fb3edaf2802fe21c1fb/Lib/test/test_argparse.py#L2560

We can define a `get_parser()` method and use `usage_when_required` and `usage_when_not_required` to check output of `parser.format_help()`.

**Review by berkerpeksag — CHANGES_REQUESTED** [review]
I just did a quick review and the PR looks good in general, thank you. We also need an entry in `Misc/NEWS` (should be in "Library" section)

**Comment by andrewnester** [comment]
@berkerpeksag thanks! just updated my PR.

**Comment by brettcannon** [comment]
To try and help move older pull requests forward, we are going through and backfilling 'awaiting' labels on pull requests that are lacking the label. Based on the current reviews, the best we can tell in an automated fashion is that a core developer requested changes to be made to this pull request.

If/when the requested changes have been made, please leave a comment that says, `I have made the requested changes; please review again`. That will trigger a bot to flag this pull request as ready for a follow-up review.

**Comment by flavianh** [comment]
Recreated at https://github.com/python/cpython/pull/14976

**Comment by aeros167** [comment]
Since this has not received a response from the author since the ``awaiting changes`` label was added on Feb 20, 2018 and was recreated at https://github.com/python/cpython/pull/14976, I'll close this PR.

**Pull request closed without merging**